File: arrays-2.py
# Dutch National Flag problem
# https://leetcode.com/problems/sort-colors
# low,mid,high = 0,0,len(arr)-1
# if arr[mid] equals to 0 then arr[low] and arr[mid] are swapped and mid,low pointers are incremented
# if arr[mid] equals to 1 no swapping only mid pointer is incremented
# if arr[mid] equals to 2 arr[high] and arr[mid] are swapped and high pointer is incremented
def sortColors(self, arr: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """ 
        low = 0 
        high = len(arr) - 1 
        mid = 0 
        while mid <= high: 
            if arr[mid] == 0: 
                arr[low],arr[mid] = arr[mid],arr[low] 
                low = low + 1 
                mid = mid + 1 
            elif arr[mid] == 1: 
                mid = mid + 1 
            else: 
                arr[mid],arr[high] = arr[high],arr[mid] 
                high = high - 1 

File: arrays-2.c
// this code is pretty stupid could wave off the laziness and write a function for swapping ughh python tuples are way better
void sortColors(int* nums, int numsSize){
    int low=0,mid=0,high=numsSize-1;
    while(mid <= high){
        if(nums[mid] == 0){
            int temp;
            temp = nums[low];
            nums[low] = nums[mid];
            nums[mid] = temp;
            low++;
            mid++;
        }
        else if(nums[mid] == 1){
            mid++;
        }
        else{
            int temp2;
            temp2 = nums[high];
            nums[high] = nums[mid];
            nums[mid] = temp2;
            high--;
        }
    }
}
